CBLClient/Apps/CBLTestServer-C/src/DatabaseMethods.cpp

```diff
@@ -446,4 +450,38 @@ namespace database_methods {
         });
 #endif
     }
+
+//names of all scopes in the database
+void database_allScope(json& body, mg_connection* conn) {
+    with<CBLDatabase *>(body,"database",[conn](CBLDatabase* db){
+        json scopes(0, nullptr);
+        CBLError* err = new CBLError();
+        auto scopeNames = CBLDatabase_ScopeNames(db, err);
+        if(err->code!=0)
+            write_serialized_body(conn, err->code);
```

Comment:
Write the code here but the other places write message (or throws or use TRY in a few places). I think it is better to have a consistent way to return an errror.